src/main/python/cfn_sphere/aws/cfn.py

```diff
@@ -303,6 +352,43 @@ def _update_stack(self, stack):
         self.client.update_stack(**kwargs)
 
     @with_boto_retry()
+    def _describe_stack_change_set(self, change_set):
+        while True:
+            resp = self.client.describe_change_set(ChangeSetName=change_set['Id'])
+
+            if resp['Status'] == 'CREATE_PENDING':
+                time.sleep(5)
+            else:
+                break
+        
+        if resp['Status'] == "FAILED":
+            print("Changeset failed with reason:", resp["StatusReason"])
+        else:
+            changset_string = get_pretty_changeset_string(resp['Changes'])
+            self.logger.info(
+                "Stack changeset with changes:\n{}".format(changset_string))
+            print(change_set['Id'])
+
+    @with_boto_retry()
+    def _create_stack_change_set(self, stack):
+        kwargs = {
+            "StackName": self.get_stack_description(stack.name)['StackId'],
+            "TemplateBody": stack.template.get_template_json(),
+            "Parameters": stack.get_parameters_list(),
+            "Capabilities": [
+                'CAPABILITY_IAM',
+                'CAPABILITY_NAMED_IAM'
+            ],
+            "ChangeSetName": stack.name + ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(5))
```

Comment:
Might get confusing with multiple changesets in play. Possible to add changeset name as a CLI argument?

Jenkins could then use the branch name or similar if that makes sense.

Reply:
Not sure on this one, sort of need to see how we end up using the commands. My thinking was there shouldn't really be any interaction with the name directly, and the need to avoid duplication.